src/cache/cache.rs

```diff
@@ -203,9 +203,31 @@ pub trait Storage: Send + Sync {
 
 /// Get a suitable `Storage` implementation from the environment.
 pub fn storage_from_environment() -> Box<Storage> {
+    let endpoint_env_res = env::var("SCCACHE_ENDPOINT");
+    let mut endpoint = String::from("s3.amazonaws.com");
```

Comment:
I tend to prefer chaining with `and_then` or `map` on `Result` and `Option` values instead of using `mut` variables, so I would probably write this like:

```rust
let endpoint = env::var("SCCACHE_ENDPOINT").unwrap_or_else(|_| String::from("s3.amazonaws.com"))
```

And similarly for the region bits below. Also: you might as well move this whole block inside the `SCCACHE_BUCKET` block, since these variables aren't useful if we're not using S3 storage.